edi-module/edi_reader.bal

```diff
@@ -0,0 +1,44 @@
+import ballerina/io;
+
+type EDIContext record {|
+    EDISchema schema;
+    int rawIndex = 0;
+    string[] rawSegments = [];
+|};
+
+# Reads EDI texts into JSON according to the given schema.
+# Each EDI document type (e.g. X12 834) can have a separate EDIReader initialized with its schema.
+public class EDIReader {
```

Comment:
Are these modeled as classes (as opposed to a function that also takes the schema) so that the reader can be reused?

Reply:
Yes, reader can be initialized with a schema once, and (re)used to read multiple EDI files of that schema.